apps/server/src/shared/domain/rules/task.rule.spec.ts

```diff
@@ -29,138 +25,207 @@ describe('TaskRule', () => {
-			const course = courseFactory.build({ teachers: [user] });
-			const lesson = lessonFactory.build({ course, hidden: true });
-			entity = taskFactory.build({ course, lesson });
-			const res = service.hasPermission(user, entity, { action: Actions.read, requiredPermissions: [] });
-			expect(res).toBe(true);
-		});
-
-		it('should return "false" if user has not permission', () => {
-			entity = taskFactory.build({ creator: user });
-			const res = service.hasPermission(user, entity, { action: Actions.read, requiredPermissions: [permissionC] });
-			expect(res).toBe(false);
-		});
-
-		it('should return "false" if user has not access to entity', () => {
-			entity = taskFactory.build();
-			const res = service.hasPermission(user, entity, { action: Actions.read, requiredPermissions: [permissionC] });
-			expect(res).toBe(false);
-		});
-	});
-
-	describe('User [STUDENT]', () => {
-		let student: User;
-		let role: Role;
-		beforeEach(() => {
-			role = roleFactory.build({ permissions: [permissionA, permissionB], name: RoleName.STUDENT });
-			student = userFactory.build({ roles: [role] });
+	describe('[method] hasPermission', () => {
+		describe('when user with permission A,B and empty task', () => {
+			const setup = () => {
+				const role = roleFactory.build({ permissions: [permissionA, permissionB] });
+				const user = userFactory.build({ roles: [role] });
+				const task = taskFactory.build();
+
+				return { role, user, task };
+			};
+
+			it('should return "false" if user has not access to task', () => {
+				const { user, task } = setup();
+				const res = service.hasPermission(user, task, { action: Actions.read, requiredPermissions: [permissionC] });
+				expect(res).toBe(false);
+			});
 		});
-		it('should return "true" if user is creator', () => {
-			entity = taskFactory.build({ creator: user });
-			const res = service.hasPermission(user, entity, { action: Actions.read, requiredPermissions: [] });
-			expect(res).toBe(true);
+		describe('when user is task creator with permission A,B but not C', () => {
+			const setup = () => {
+				const role = roleFactory.build({ permissions: [permissionA, permissionB] });
+				const user = userFactory.build({ roles: [role] });
+				const task = taskFactory.build({ creator: user });
+
+				return { role, user, task };
+			};
+
+			it('should call baseRule.hasAllPermissions', () => {
+				const { user, task } = setup();
+				const spy = jest.spyOn(service.utils, 'hasAllPermissions');
+				service.hasPermission(user, task, { action: Actions.read, requiredPermissions: [] });
+				expect(spy).toBeCalledWith(user, []);
+			});
+
+			it('should call baseRule.hasAccessToEntity', () => {
+				const { user, task } = setup();
+				const spy = jest.spyOn(service.utils, 'hasAccessToEntity');
+				service.hasPermission(user, task, { action: Actions.read, requiredPermissions: [] });
+				expect(spy).toBeCalledWith(user, task, ['creator']);
+			});
+
+			it('should return "true" if user is creator', () => {
```

Comment:
user is creator is already stated in the description
should retrurn true when no required permissions are needed
and perhaps another test should return true when user has all required permissions

Reply:
I would not want to introduce new test.
We can do that in a next ticket.